app/models/manageiq/providers/datawarehouse_manager.rb

```diff
@@ -1,5 +1,8 @@
 module ManageIQ::Providers
   class DatawarehouseManager < BaseManager
+    has_many :datawarehouse_nodes, :foreign_key => :ems_id, :dependent => :destroy
+    has_many :advanced_settings, :as => :resource, :dependent => :destroy
```

Comment:
Why did you choose to use advanced_settings and not custom_attributes?
Some features are not yet supported by advanced settings, Such as access from reports and expressions.
Note this comment: https://github.com/ManageIQ/manageiq/pull/12924#discussion_r97896303